syntax = "proto3";

service MasterNode{
    rpc Store(StoreRequest) returns (StoreReply){}
    rpc Read(ReadRequest) returns (ReadReply){}    
}

message StoreRequest{
    string file_name = 1;  
    bytes file_content = 2;
    string timestamp = 3;
    string user_id = 4;
}

message StoreReply{
    string reply_msg = 1;
    bool success  = 2;    
}

message ReadRequest{
    string file_name = 1;
    string timestamp = 2;
}

message ReadReply{
    bytes reply_file = 1;
    bool success = 2;  
}
